otx/algorithms/detection/tasks/openvino.py

```diff
@@ -340,18 +341,47 @@ def infer(
                 dataset_item.append_metadata_item(representation_vector, model=self.model)
 
             if add_saliency_map and saliency_map is not None:
-                actmap = get_actmap(saliency_map, (dataset_item.width, dataset_item.height))
-                saliency_map_media = ResultMediaEntity(
-                    name="Saliency Map",
-                    type="saliency_map",
-                    annotation_scene=dataset_item.annotation_scene,
-                    numpy=actmap,
-                    roi=dataset_item.roi,
+                add_saliency_maps_to_dataset_item(
+                    dataset_item=dataset_item,
+                    saliency_map=saliency_map,
+                    model=self.model,
+                    labels=self.task_environment.get_labels(),
+                    task="det",
+                    predicted_scene=predicted_scene,
                 )
-                dataset_item.append_metadata_item(saliency_map_media, model=self.model)
         logger.info("OpenVINO inference completed")
         return dataset
 
+    @check_input_parameters_type({"dataset": DatasetParamTypeCheck})
+    def explain(
+        self,
+        dataset: DatasetEntity,
+        explain_parameters: Optional[InferenceParameters] = None,
+    ) -> DatasetEntity:
+        """Explain function of OpenVINODetectionTask."""
+        logger.info("Start OpenVINO explain")
+
+        update_progress_callback = default_progress_callback
+        if explain_parameters is not None:
+            update_progress_callback = explain_parameters.update_progress  # type: ignore
+        dataset_size = len(dataset)
+        for i, dataset_item in enumerate(dataset, 1):
+            predicted_scene, features = self.inferencer.predict(dataset_item.numpy)
+            dataset_item.append_annotations(predicted_scene.annotations)
+            update_progress_callback(int(i / dataset_size * 100), None)
```

Comment:
May be it is fair to move this progress callback to the bottom of the fool loop body?

Reply:
Right, fixed